### PR TITLE
[MB-1829] Fix reroute/restore expired messages from DLC

### PR DESCRIPTION
**Pull request by sirojan24**
Fix the NPE when to reroute/restore the expired messages from DLC

Associated JIRA Link: https://wso2.org/jira/browse/MB-1829
Merge with https://github.com/wso2/carbon-business-messaging/pull/364

### CODE REVIEW

File: modules/andes-core/broker/src/main/java/org/wso2/andes/server/information/management/QueueManagementInformationMBean.java [hunk 1]
```diff
@@ -424,26 +425,26 @@ public void restoreMessagesFromDeadLetterQueue(@MBeanOperationParameter(name = "
...
+                        metadata.setStorageQueueName(AndesUtils.getStorageQueueForDestination(destination,
+                                AMQPUtils.DIRECT_EXCHANGE_NAME,destination,true));
+                        messagesToRemove.add(metadata);
+                        AndesMessageMetadata clonedMetadata = metadata.shallowCopy(metadata.getMessageID());
+                        AndesMessage andesMessage = new AndesMessage(clonedMetadata);
+                        // Update Andes message with all the chunk details
+                        List<AndesMessagePart> messageParts = messageContent.get(messageId);
+                        for (AndesMessagePart messagePart : messageParts) {
+                            andesMessage.addMessagePart(messagePart);
+                        }
```
Comment (a5anka) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/server/information/management/QueueManagementInformationMBean.java]: Look like these set of code used to created the andesMessage is repeated in the other modified method. Can we use a private method for that.

File: modules/andes-core/broker/src/main/java/org/wso2/andes/server/information/management/QueueManagementInformationMBean.java [hunk 2]
```diff
@@ -457,7 +458,7 @@ public void restoreMessagesFromDeadLetterQueue(@MBeanOperationParameter(name = "
                 throw new RuntimeException("Error restoring messages from " + destinationQueueName, e);
             }
         }
-
+        return unavailableMessageCount;
```
Comment (a5anka) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/server/information/management/QueueManagementInformationMBean.java]: Why do we need to return the unavailable count?

File: modules/andes-core/broker/src/main/java/org/wso2/andes/server/information/management/QueueManagementInformationMBean.java (hunk 1 above)
Comment (sirojan24) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/server/information/management/QueueManagementInformationMBean.java]: A private method is added in d2f85264be41e05fc47b66ab177b7102df8ea202

File: modules/andes-core/broker/src/main/java/org/wso2/andes/server/information/management/QueueManagementInformationMBean.java [hunk 3]
```diff
@@ -495,30 +490,26 @@ public void restoreMessagesFromDeadLetterQueue(@MBeanOperationParameter(name = "
...
+                        metadata.setStorageQueueName(newDestinationQueueName);
+                        metadata.setMessageRouterName(newStorageQueue.getMessageRouter().getName());
+                        metadata.updateMetadata(newDestinationQueueName, newStorageQueue.getMessageRouter().getName());
+                        messagesToRemove.add(metadata);
+                        // get the message chunk details
+                        List<AndesMessagePart> messageParts = messageContent.get(messageId);
+                        AndesMessage andesMessage = createMessage(metadata, messageParts);
+                        // Handover message to Andes. This will generate a new message ID and store it
+                        Andes.getInstance().messageReceived(andesMessage, andesChannel, disablePubAck);
+                    }else{
```
Comment (sdkottegoda) [inline modules/andes-core/broker/src/main/java/org/wso2/andes/server/information/management/QueueManagementInformationMBean.java]: Need a space in between

**Pull request merged**